pages/extensions/amp_dependencies/__init__.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+import uuid
+from sets import Set
+
+from grow import extensions
+from grow.documents import document, static_document
+from grow.extensions import hooks
+
+class AmpDependencies(object):
+    def __init__(self, doc):
+        self._doc = doc
+        self._pod = doc.pod
+        # Used to determine where to print all dependencies
+        self._placeholder = '{}'.format(uuid.uuid4())
```

Comment:
There will be only one instance of AmpDependencies for a document.
So we do not need a dynamic placeholder

Reply:
Correct! Changed it to a static placeholder.